src/Schema/Index.php

```diff
@@ -334,6 +344,16 @@ public function getOptions()
         return $this->options;
     }
 
+    /**
+     * Returns the previous (quoted) index name or the current one if unchanged
+     *
+     * @return string Previous index name or current index name
+     */
+    public function getPreviousName()
```

Comment:
It's not the responsibility of a value object to contain its previous name. It should be the responsibility of the object that renames it.

Reply:
That's now a priciple decision how we fix that:
- In https://github.com/doctrine/dbal/pull/4989 we solve the problem in the platform domain
- This PR tries to solve the problem in the schema domain and needs the name of the previous (quoted) index name
- The last possibility is to add an IndexDiff class like for tables but that requires much more changes in the code